pkg/controller/gittrack/gittrack_controller.go

```diff
@@ -134,12 +146,14 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	err = c.Watch(&source.Kind{Type: &farosv1alpha1.ClusterGitTrackObject{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &farosv1alpha1.ClusterGitTrack{},
-	})
-	if err != nil {
-		return err
+	if opts.clusterGitTrackMode != farosflags.CGTMDisabled {
```

Comment:
Why not put this in the same if statement? Seems odd to evaluate it multiple times?